### PR TITLE
MockSpyLog

### PR DESCRIPTION
**Pull request by Yanetta**

### CODE REVIEW

File: src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <xml status="WARN" monitorInterval="30">
 	<Properties>
 		<Property name="LOG_PATTERN">
-			%-5p | %d{yyyy-MM-dd HH:mm:ss} | [%t] %C{2}	(%F:%L) -- %m%n
```
Comment (Cherkashyn) [inline src/main/resources/log4j2.xml:6]: good

File: src/test/java/FruitShopMockSpy.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+@RunWith(MockitoJUnitRunner.class)
+public class FruitShopMockSpy {
+
+    @Spy
+    @InjectMocks
+    private FruitShop fruitShop;
+    @Mock
+    private List<Fruit> fruits;
+    @Mock
+    private Fruit apple;
```
Comment (Cherkashyn) [inline src/test/java/FruitShopMockSpy.java:35]: Why you are creating mock on all Fruit objects? You need to mock just one object Fruit fruit, and create a different realizations of it methods.

File: src/test/java/FruitShopMockSpy.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    private void createFruitMock(Fruit fruit, FruitType fruitType, LocalDate date, int timeTolive) {
+        when(fruit.getDeliveryDate()).thenReturn(date);
+        when(fruit.getDayToLive()).thenReturn(timeTolive);
+        when(fruit.clone()).thenReturn(fruit);
+        when(fruit.getFruitType()).thenReturn(fruitType);
+    }
+
+    @Test
+    public void testAllFresh() {
```
Comment (Cherkashyn) [inline src/test/java/FruitShopMockSpy.java:60]: You need to "override" methods of Fruit class here, not simple objects.

File: src/test/java/FruitShopMockSpy.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        List<Fruit> expected = new ArrayList<>();
+        expected.add(apple);
+        expected.add(pear);
+        List<Fruit> actual = fruitShop.allFresh(LocalDate.of(2019, Month.FEBRUARY, 20));
+        verify(apple, times(1)).clone();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testAllFruitOfFruitTypeApple() {
```
Comment (Cherkashyn) [inline src/test/java/FruitShopMockSpy.java:70]: method name is good

**Review by Cherkashyn — APPROVED** [review]

File: src/test/java/FruitShopMockSpy.java (hunk 2 above)
Comment (SerhiiPasko) [reply src/test/java/FruitShopMockSpy.java:35]: It is normal. More bigger level (List in our case) also can be mock

File: src/test/java/FruitShopMockSpy.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    @Before
+    public void setUp() {
+        createFruitMock(apple, FruitType.APPLE, LocalDate.of(2019, Month.JANUARY, 15), 55);
+        createFruitMock(pear, FruitType.PEAR, LocalDate.of(2019, Month.JANUARY, 16), 45);
+        createFruitMock(strawberry, FruitType.STRAWBERRY, LocalDate.of(2018, Month.DECEMBER, 6), 60);
+        createFruitMock(orange, FruitType.ORANGE, LocalDate.of(2019, Month.JANUARY, 10), 40);
+        when(fruitShop.getFruits()).thenReturn(Arrays.asList(new Fruit[]{apple, pear, strawberry, orange}));
+    }
+
+    private void createFruitMock(Fruit fruit, FruitType fruitType, LocalDate date, int timeTolive) {
```
Comment (SerhiiPasko) [inline src/test/java/FruitShopMockSpy.java:52]: I see my code =)

**Review by SerhiiPasko — APPROVED** [review]
Very good

**Pull request merged**